awsrds/db_instance.go

```diff
@@ -45,6 +45,47 @@ func (ct ByCreateTime) Less(i, j int) bool {
 	return aws.TimeValue(ct[i].SnapshotCreateTime).After(aws.TimeValue(ct[j].SnapshotCreateTime))
 }
 
+type awsRdsErr struct {
+	orig error
+	code string
+}
+
+func (a awsRdsErr) Error() string {
+	return a.orig.Error()
+}
+
+func (a awsRdsErr) Code() string {
+	return a.code
+}
+
+func (a awsRdsErr) OrigErr() error {
+	return a.orig
+}
+
+func NewError(err error, code string) Error {
+	return &awsRdsErr{
+		orig: err,
+		code: code,
+	}
+}
```

Comment:
Would it make sense to use `New` from https://docs.aws.amazon.com/sdk-for-go/api/aws/awserr/ instead of having our own?

This might not make sense, I feel like my head is going around in circles about it

Reply:
I think if anything, using a wrapped error would make more sense, but both of them will result in `broker.go` having to do some sniffing to figure out what this `error` it's been handed actually is, and that to me is a bit of an abstraction violation.

Though, it's not like `broker.go` doesn't already violate the abstraction. Else it wouldn't have to import `github.com/aws/aws-sdk-go/service/rds` up top.

Reply:
* sighs * - it's a bit of a rabbit hole

Reply:
Nobody seems to have a strong opinion on this so I'm just mergin'